src/ray/raylet/scheduling_policy.cc

```diff
@@ -63,12 +60,18 @@ std::unordered_map<TaskID, ClientID> SchedulingPolicy::Schedule(
       // TODO(atumanov): change uniform random to discrete, weighted by resource capacity.
       std::uniform_int_distribution<int> distribution(0, client_keys.size() - 1);
       int client_key_index = distribution(gen_);
-      decision[task_id] = client_keys[client_key_index];
+      const ClientID &dst_client_id = client_keys[client_key_index];
+      // Decrement destination client's available resources at placement decision time.
+      cluster_resources[dst_client_id].Acquire(
+          t.GetTaskSpecification().GetRequiredResources());
+      // Fractional and enumerated resources will be acquired on task dispatch.
+      decision[task_id] = dst_client_id;
       RAY_LOG(DEBUG) << "[SchedulingPolicy] idx=" << client_key_index << " " << task_id
                      << " --> " << client_keys[client_key_index];
     } else {
-      // There are no nodes that can feasibily execute this task. TODO(rkn): Propagate a
-      // warning to the user.
+      // There are no nodes that can feasibly execute this task. The task remains
+      // placeable until cluster capacity becomes available.
+      // TODO(rkn): Propagate a warning to the user.
       RAY_LOG(WARNING) << "This task requires "
```

Comment:
If I understand this correctly, is this warning now getting logged if there are no clients with enough *available* vs *total* resources? Printing the warning in the former case seems like it may produce a lot of spurious messages, especially since we'll now be calling the `Schedule` method on the policy with resource maps that only have one client.